packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/InAppPurchasePluginTests.m

```diff
@@ -708,12 +751,20 @@ - (void)testHandleTransactionsUpdated {
 
   InAppPurchasePlugin *plugin = [[InAppPurchasePluginStub alloc]
       initWithReceiptManager:self.receiptManagerStub
-              handlerFactory:^FIAPRequestHandler *(SKRequest *request) {
-                return [[FIAPRequestHandler alloc] initWithRequest:request];
+              handlerFactory:^DefaultRequestHandler *(SKRequest *request) {
+                return [[DefaultRequestHandler alloc]
+                    initWithRequestHandler:[[FIAPRequestHandler alloc] initWithRequest:request]];
               }];
-  FlutterMethodChannel *mockChannel = OCMClassMock([FlutterMethodChannel class]);
-  plugin.transactionObserverCallbackChannel = mockChannel;
-  OCMStub([mockChannel invokeMethod:[OCMArg any] arguments:[OCMArg any]]);
+  TestMethodChannel *testChannel = [[TestMethodChannel alloc] init];
+  __block NSInteger invokeMethodCount = 0;
+
+  testChannel.invokeMethodChannelStub = ^(NSString *_Nonnull method, id _Nonnull arguments) {
+    XCTAssertEqualObjects(@"updatedTransactions", method);
+    XCTAssertNotNil(arguments);
+    invokeMethodCount++;
+  };
+
+  plugin.transactionObserverCallbackChannel = testChannel;
```

Comment:
This isn't a good practice. Can you add a TODO to inject the correct channel, rather than the existing practice here (which init it using the real channel only to be overwritten to the test channel).

Reply:
done